pkg/docker/storage.jsx

```diff
@@ -0,0 +1,512 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+define([
+    "jquery",
+    "base1/cockpit",
+    "docker/react",
+    "docker/cockpit-components-dialog",
+    "data!docker/config.json"
+], function($, cockpit, React, dialog_view, config_data) {
+    var _ = cockpit.gettext;
+    var C_ = cockpit.gettext;
+
+    var config = JSON.parse(config_data);
+
+    console.log(config);
+
+    function init_model(callback) {
+        var pool_devices = [ ];
+
+        var self = {
+            error: null,
+            driver: "",
+            pool_devices: [ ],
+            extra_devices: [ ],
+            total: 0,
+            used: 0
+        };
+
+        function cmp_drive(a, b) {
+            return a.sort_index - b.sort_index;
+        }
+
+        cockpit.spawn([ config.cockpit_atomic_storage, "monitor" ],
```

Comment:
When do we kill this process? Should we kill it when the user hides the page? See #4492